geoutils/satimg.py

```diff
@@ -314,4 +315,20 @@ def __parse_metadata_from_fn(self,silent=False):
     def __parse_metadata_from_file(self,fn_meta):
         pass
 
+    def __copy__(self):
```

Comment:
This will have the exact same problem as I mentioned in xdem:

> A test should be added to assert the consistency of the `.data` attribute.
>
> A quick version would be:
> ```python
> dem1 = DEM("path/to/file")
> dem1.data += 5
>
> dem2 = dem1.copy()
>
>assert np.mean(np.abs(dem1.data - dem2.data)) < 0.001
> ```
> And I suspect the above test will fail right now.

_Originally posted by @erikmannerfelt in https://github.com/GlacioHack/xdem/pull/46#discussion_r592968427_

The problem is that loading from the `filename` attribute will not retain intermediate changes to the `data` attribute.

Reply:
I rather suggest something along:

```python
def copy(self):
    img_copy = self.super().copy()
    # copy all the SatelliteImage attributes somehow, maybe with a __dict__comparison
    for key in self.__dict__:
           if key in img_copy.__dict__:
               continue
           img_copy.__dict__[key] = self.__dict__[key]
```
I don't know if the above will work though, because it may be a Raster instance with SatelliteImage methods and attributes!